### DISCUSSION TITLE
Is it possible to define multiple RSA host private keys in ipsec.secrets?

### DISCUSSION THREAD

**Discussion by acrolink** [strongswan/strongswan#568] [Q&A]
I understand that I can point to the host's private key like this:

`: RSA vpnHostKey_serverA.pem` inside `ipsec.secrets`

and in `ipsec.conf` I can define the corresponding certificate like this:

`leftcert=vpnHostCert_serverA.pem`

My question, is it possible to define and use multiple `RSA` private keys? And then make two or more groups of clients, each group with dedicated `RSA` key?

Thank you.

**Comment by Thermi** [comment]
Yes and no.

Yes, you can define and use multiple private keys.

Private keys always correspond to one (or several? I'd need to check the code for the latter) public key/certificate.

Because having a private key for a public key is used to prove "ownership" of it but is not required to verify ownership, it makes no sense to store a private key for an initiator's public key on the responder. Or vice versa.
Only the owner of a certificate requires the private key for it.
If anybody else has the private key, the certificate is useless because a different entity than the owner can prove ownership of the certificate.

**Comment by acrolink** [comment]
I understand. But practically how to specify more than one `RSA` key inside `ipsec.secrets` ? Is that possible?

I tried like this:

```
@a.mydomain.com : RSA vpnHost_ServerA.pem
@b.mydomain.com : RSA vpnHost_ServerB.pem
```

But seems that the value before `:` does not make any difference.

**Reply by Thermi** [reply]
private keys do not have a selector.

**Reply by Thermi** [reply]
Just add several lines, one for each private key.
